From f310e79920d36017f25eed119190a51613c49a2f Mon Sep 17 00:00:00 2001
From: Tatsuya Matsuo <96454784+matsuo-tat@users.noreply.github.com>
Date: Fri, 15 Dec 2023 16:06:00 +0900
Subject: [PATCH 1/3] [sonic-cfggen] Add --use-unix-sock option

This change adds an option for control whether to use Unix domain socket or
TCP for redis connection.
---
 src/sonic-config-engine/sonic-cfggen | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/src/sonic-config-engine/sonic-cfggen b/src/sonic-config-engine/sonic-cfggen
index 287640d8a..858cf94d9 100755
--- a/src/sonic-config-engine/sonic-cfggen
+++ b/src/sonic-config-engine/sonic-cfggen
@@ -267,6 +267,8 @@ def main():
     parser.add_argument("-a", "--additional-data", help="addition data, in json string")
     parser.add_argument("-d", "--from-db", help="read config from configdb", action='store_true')
     parser.add_argument("-H", "--platform-info", help="read platform and hardware info", action='store_true')
+    parser.add_argument("--use-unix-sock", help="use unix domain socket for redis connection, if not specified use tcp",
+                        action='store_true')
     parser.add_argument("-s", "--redis-unix-sock-file", help="unix sock file for redis connection")
     group = parser.add_mutually_exclusive_group()
     group.add_argument("-t", "--template", help="render the data with the template file", action="append", default=[],
@@ -284,6 +286,7 @@ def main():
     platform = device_info.get_platform()
 
     db_kwargs = {}
+    use_unix_sock = args.use_unix_sock
     if args.redis_unix_sock_file is not None:
         db_kwargs['unix_socket_path'] = args.redis_unix_sock_file
 
@@ -358,7 +361,6 @@ def main():
         deep_update(data, json.loads(args.additional_data))
 
     if args.from_db:
-        use_unix_sock = True if os.getuid() == 0 else False
         if args.namespace is None:
             configdb = ConfigDBPipeConnector(use_unix_socket_path=use_unix_sock, **db_kwargs)
         else:
@@ -432,10 +434,10 @@ def main():
 
     if args.write_to_db:
         if args.namespace is None:
-            configdb = ConfigDBPipeConnector(use_unix_socket_path=True, **db_kwargs)
+            configdb = ConfigDBPipeConnector(use_unix_socket_path=use_unix_sock, **db_kwargs)
         else:
             SonicDBConfig.load_sonic_global_db_config(namespace=args.namespace)
-            configdb = ConfigDBPipeConnector(use_unix_socket_path=True, namespace=args.namespace, **db_kwargs)
+            configdb = ConfigDBPipeConnector(use_unix_socket_path=use_unix_sock, namespace=args.namespace, **db_kwargs)
 
         configdb.connect(False)
         configdb.mod_config(FormatConverter.output_to_db(data))
-- 
2.20.1

